client/components/IndividualChallenge.js

```diff
@@ -13,27 +17,36 @@ const _IndividualChallenge = props => {
     console.log({ html, css, js });
   };
 
-  const putValue = isSubmit => {
+  const updateValue = isSubmit => {
     const userAnswer = { html, css, js, submitted: true, challengeId: props.id };
     props
-      .putUserchallenge(userAnswer, props.id, isSubmit)
+      .updateUserchallenge(userAnswer, props.id, isSubmit)
       .then(userchallenge => console.log(userchallenge))
       .catch(ex => console.log(ex));
   };
 
-  const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
   const options = {
     lineNumbers: true,
     mode: 'javascript',
   };
+
+  if (!Object.keys(props.challenge).length) {
```

Comment:
I think this should be written out, it's a little hard to read as is.

`if (Object.keys(props.challenge).length === 0) {`